src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -0,0 +1,526 @@
+                    {
+                        // ArgumentException's message is first. If it's not null nor "", it's "meaningful".
+                        return !IsNullOrEmptyMessage(args[0]);
+                    }
+
+                    // The other exceptions all have the message second. If they only have 0 or 1 arguments,
+                    // there's no message, and if they have more than one, it's again "meaningful" if it's not null nor "".
+                    return args.Length >= 2 && !IsNullOrEmptyMessage(args[1]);
+
+                    static bool IsNullOrEmptyMessage(IArgumentOperation arg) =>
+                        arg.Value.WalkDownConversion() is ILiteralOperation { ConstantValue.HasValue: true } literal &&
+                        literal.ConstantValue.Value is null or "";
+                }
+            });
+        }
+
+        /// <summary>Gets the <see cref="IParameterReferenceOperation"/> being null checked by the condition, or else null.</summary>
+        private static bool IsParameterNullCheck(IOperation condition, [NotNullWhen(true)] out IParameterReferenceOperation? parameterReference)
+        {
+            parameterReference = null;
+
+            if (condition is IIsPatternOperation isPattern &&
+                isPattern.Pattern is IConstantPatternOperation { Value.ConstantValue: { HasValue: true, Value: null } })
+            {
+                // arg is null
+                parameterReference = isPattern.Value as IParameterReferenceOperation;
+            }
+            else if (condition is IBinaryOperation { OperatorKind: BinaryOperatorKind.Equals } equalsOp)
+            {
+                if (equalsOp.RightOperand.HasNullConstantValue())
+                {
+                    // arg == null
+                    parameterReference = equalsOp.LeftOperand.WalkDownConversion() as IParameterReferenceOperation;
+                }
+                else if (equalsOp.LeftOperand.HasNullConstantValue())
+                {
+                    // null == arg
+                    parameterReference = equalsOp.RightOperand.WalkDownConversion() as IParameterReferenceOperation;
+                }
+            }
+
+            return parameterReference is not null;
+        }
+
+        /// <summary>Gets the string <see cref="IParameterReferenceOperation"/> being checked by the condition for being null or empty, or else null.</summary>
+        private static bool IsNullOrEmptyCheck(ISymbol stringIsNullOrEmpty, ISymbol stringLength, ISymbol stringEmpty, IOperation condition, [NotNullWhen(true)] out IParameterReferenceOperation? parameterReference)
+        {
+            if (condition is IInvocationOperation invocationOperation)
+            {
+                // (string.IsNullOrEmpty(arg))
+                if (stringIsNullOrEmpty is not null &&
+                    SymbolEqualityComparer.Default.Equals(invocationOperation.TargetMethod, stringIsNullOrEmpty) &&
```

Comment:
We already return very early if it's null. In addition, SymbolEqualityComparer.Default already handles null properly.

```suggestion
                if (SymbolEqualityComparer.Default.Equals(invocationOperation.TargetMethod, stringIsNullOrEmpty) &&
```